### DISCUSSION TITLE
Updating Mask Template File

### DISCUSSION THREAD

**Discussion by Zak-Eckert** [LabeoTech/Umit#36] [General]
Greetings, 
When working on a longitudinal sample, the window can often become damaged or foggy with time. I have noticed that this creates issues in the automatic allignment for overlapping images, although bregma can still be seen. 

What would you recommend the best course of action be? Generating a new mask file for the later timepoints (if so, how would this be done)? Doing manual alignment (perferably not)? Or another course of action. 

Thanks!
Zak

**Comment by BrunoOFSouza** [comment]
Hi Zak, 

Indeed, the current alignment algorithm can perform poorly in such cases. 
In your case, what is the proportion of failed alignment that you have? Does the automatic alignment fails on most of the data? 
If it is a minority, I think that you will save time with the manual alignment on the long run because creating ROIs could be more time-consuming (I guess...).

Perhaps it could be the case to review the code and try to make improvements. 
Can you attach some sample data here? 
For instance: 
* The Reference Frame (the ***ImagingReferenceFrame.mat*** file)
* One good alignment*
* One bad alignment*

 *Just the data from one frame for each one. 
______

Here is  a code snippet on how to extract a frame from an Image time series in Matlab:
```
%% Load the data:
% First, change the Matlab's current directory to the one containing the data:
cd(uigetdir)
% Load the data (in this example, we open the green channel):
datName = 'green.dat';
data = loadDatFile(datName);
% Extract the first frame
frame = data(:,:,1);
%% Save the data to a file:
% Save to a .mat file named 'greenFrame1' in the current directory:
filename = 'greenFrame1.mat'; 
save(filename, 'frame');
